zebrad/src/components/rpc.rs

```diff
@@ -0,0 +1,37 @@
+//! An RPC endpoint.
+
+use jsonrpc_core;
+
+use jsonrpc_http_server::ServerBuilder;
+
+use zebra_rpc::rpc::{Rpc, RpcImpl};
+
+pub mod config;
+pub use config::Config;
+
+/// Zebra RPC Server
+pub struct RpcServer {}
+
+impl RpcServer {
+    /// Start a new RPC server endpoint
+    pub async fn new(config: Config) -> Self {
+        if config.listen {
+            info!("Trying to open RPC endpoint at {}...", config.listen_addr);
+
+            // Create handler compatible with V1 and V2 RPC protocols
+            let mut io =
+                jsonrpc_core::IoHandler::with_compatibility(jsonrpc_core::Compatibility::Both);
+            io.extend_with(RpcImpl.to_delegate());
+
+            let server = ServerBuilder::new(io)
+                .threads(1)
```

Comment:
Runtime Behaviour: Can we make the RPC server use `zebrad`'s tokio executor?

I think this will make shutdown more reliable, and make performance better.

```suggestion
                // use the same tokio executor as the rest of Zebra
                .event_loop_executor(tokio::runtime::Handle::current())
                .threads(1)
```

Here is the documentation explaining how I did that:
https://docs.rs/jsonrpc-http-server/18.0.0/jsonrpc_http_server/struct.ServerBuilder.html#method.threads

Here is the extra method I've added:
https://docs.rs/jsonrpc-http-server/18.0.0/jsonrpc_http_server/struct.ServerBuilder.html#method.event_loop_executor

Reply:
Panic: When I run this code, I get this panic from `tokio`:

> Cannot drop a runtime in a context where blocking is not allowed. This happens when a runtime is dropped from within an asynchronous context. 

```
Feb 21 09:33:21.514  INFO {zebrad="97c6a8e" net="Test"}: zebrad::components::rpc: Trying to open RPC endpoint at 127.0.0.1:8232...
Feb 21 09:33:21.514  INFO {zebrad="97c6a8e" net="Test"}: zebrad::components::mempool::gossip: initializing transaction gossip task
Feb 21 09:33:21.514  INFO {zebrad="97c6a8e" net="Test"}:sync: zebrad::components::sync: starting sync, obtaining new tips state_tip=Some(Height(1773895))
Feb 21 09:33:21.515  INFO {zebrad="97c6a8e" net="Test"}: zebrad::commands::start: estimated progress to chain tip sync_percent=99.994 % current_height=Height(1773895) remaining_sync_blocks=100 time_since_last_state_block=PT0S
The application panicked (crashed).
Message:  Cannot drop a runtime in a context where blocking is not allowed. This happens when a runtime is dropped from within an asynchronous context.
Location: /home/dev/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.16.1/src/runtime/blocking/shutdown.rs:51
Metadata:
version: 1.0.0-beta.4+86.g97c6a8e.modified
Zcash network: Testnet
state version: 12
branch: issue3140
git commit: 97c6a8e
commit timestamp: 2022-02-20T15:47:05+00:00
target triple: x86_64-unknown-linux-gnu
build profile: release
```